### PR TITLE
Fix parsing issue with trailing comma in destructuring

### PR DESCRIPTION
**Pull request by arv**
Also make sure the ParseTreeWriter prints trailnig holes.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/outputgeneration/ParseTreeWriter.js [hunk 1]
```diff
@@ -177,6 +181,10 @@ export class ParseTreeWriter extends ParseTreeVisitor {
   visitArrayPattern(tree) {
     this.write_(OPEN_SQUARE);
     this.writeList_(tree.elements, COMMA, false);
+    if (tree.elements[tree.elements.length - 1] === null) {
+      this.write_(COMMA);
+      this.writeSpace_();
```
Comment (johnjbarton) [inline src/outputgeneration/ParseTreeWriter.js]: Seems like we could have a test that verifies this case (and fails without this code).

**Comment by johnjbarton** [comment]
lgtm, just unclear on why there isn't a test.

File: src/outputgeneration/ParseTreeWriter.js (hunk 1 above)
Comment (arv) [inline src/outputgeneration/ParseTreeWriter.js]: The PR is missing the test... I wonder which computer I left it on?

**Pull request merged**